### PR TITLE
[Feature] Specify default fill mode for video tiles

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

https://wearezeta.atlassian.net/browse/ZIOS-13481

- Scale video to fill available space by default
- Scale screenshare video to fit available space to preserve aspect

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift [hunk 1]
```diff
@@ -63,45 +65,23 @@ final class VideoPreviewView: BaseVideoPreviewView {
...
-    override func layoutSubviews() {
-        super.layoutSubviews()
-        
-        if !userHasSetFillMode {
-            previewView?.shouldFill = (previewView?.videoSize.aspectRatio == previewView?.frame.size.aspectRatio)
-        }
+
+    override func updateFillMode() {
+        guard let previewView = previewView else { return }
+        previewView.shouldFill = shouldFill && !isScreenSharing
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift:81]: Does this mean that the user can never change the fill mode of screensharing?

Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift:81]: mh, yes

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift [hunk 2]
```diff
@@ -63,45 +65,23 @@ final class VideoPreviewView: BaseVideoPreviewView {
...
-        } else {
-            insertSubview(preview, belowSubview: userDetailsView)
-        }
-        preview.fitInSuperview()
-        preview.shouldFill = true
 
-        previewView = preview
+    // MARK: - Fill mode
+    private var isScreenSharing: Bool {
+        return stream.videoState == .screenSharing ? true : false
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift:71]: You can simplify this to

```swift
return stream.videoState == .screenSharing
```

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift [hunk 3]
```diff
@@ -63,45 +65,23 @@ final class VideoPreviewView: BaseVideoPreviewView {
...
     }
 
     public func switchFillMode() {
-        guard let previewView = previewView else { return }
-        userHasSetFillMode = true
-        previewView.shouldFill = !previewView.shouldFill
+        shouldFill = !shouldFill
+        updateFillMode()
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/VideoPreviewView.swift:76]: Perhaps it's a bit cleaner to expose `shouldFill` and use the setter to call `updateFillMode()`. The caller could then call `shouldFill.toggle()` to switch between fill modes.

**Pull request merged**